File: TransCoder_tokenized_test_set_functions/java/LENGTH_LONGEST_BALANCED_SUBSEQUENCE_1.java
static int maxLength ( String s, int n ) {
  int invalidOpenBraces = 0;
  int invalidCloseBraces = 0;
  for ( int i = 0;
  i < n;
  i ++ ) {
    if ( s . charAt ( i ) == '(' ) {
      invalidOpenBraces ++;
    }
    else {
      if ( invalidOpenBraces == 0 ) {
        invalidCloseBraces ++;
      }
      else {
        invalidOpenBraces --;
      }
    }
  }
  return ( n - ( invalidOpenBraces + invalidCloseBraces ) );
}

File: TransCoder_tokenized_test_set_functions/python/LENGTH_LONGEST_BALANCED_SUBSEQUENCE_1.py
def maxLength ( s, n ) :
    invalidOpenBraces = 0;
    
    invalidCloseBraces = 0;
    
    for i in range ( n ) :
        if ( s [ i ] == '(' ) :
            invalidOpenBraces += 1
        else :
            if ( invalidOpenBraces == 0 ) :
                invalidCloseBraces += 1
            else :
                invalidOpenBraces -= 1
    return ( n - ( invalidOpenBraces + invalidCloseBraces ) )
